back/lib/tasks/single_use/services/file_upload_custom_field_migration_service.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class Tasks::SingleUse::Services::FileUploadCustomFieldMigrationService
+  def initialize
+    @stats = { ideas: 0, file_fields: 0, ideas_updated: 0, errors: [] }
+  end
+
+  attr_reader :stats
+
+  def migrate(persist_changes)
+    Rails.logger.info 'Migrating file upload format for idea custom fields.'
+    Rails.logger.info "Persist: #{persist_changes}"
+
+    custom_field_keys = CustomField.where(input_type: 'file_upload').pluck(:key)
+    ideas = Idea.where.not(custom_field_values: {})
+    @stats[:ideas] = ideas.count
+
+    ideas.each do |idea|
+      Rails.logger.info "MIGRATING IDEA: #{idea.id}."
+      save_idea = false
+      idea.custom_field_values.each do |key, value|
+        next unless custom_field_keys.include? key
+
+        file = IdeaFile.find_by(id: value)
+        next unless file
+
+        idea.custom_field_values[key] = { id: file.id, name: file.name }
+        save_idea = true
+        @stats[:file_fields] += 1
+        Rails.logger.info "TRANSFORMED FILE: #{file.id} - #{file.name}"
+      end
+
+      if save_idea && persist_changes
+        # TODO: JS - Is there any impact on the updated_at dates changing?
```

Comment:
According to Vscode's "Find all references", `updated_at` is not used https://github.com/CitizenLabDotCo/citizenlab/blob/4b378e48a272bd1dd14379cb6c80a9d68d3e9d7a/front/app/api/ideas/types.ts#L97-L98